src/parser/position.rs

```diff
@@ -0,0 +1,77 @@
+use crate::parser::Span;
+#[cfg(not(target_arch = "wasm32"))]
+use proptest::prelude::*;
+
+/// Source code position of an expression in a file
+#[derive(Clone, Copy, Debug)]
+pub enum Pos {
+    No,
+    Pos {
+        from_offset: usize,
+        from_line: usize,
+        from_column: usize,
+        upto_offset: usize,
+        upto_line: usize,
+        upto_column: usize,
+    },
+}
+
+impl PartialEq for Pos {
+    fn eq(&self, _other: &Self) -> bool {
+        true
+    }
+}
+
+impl Eq for Pos {}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl Arbitrary for Pos {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        any::<()>().prop_map(|_| Pos::No).boxed()
+    }
+}
```

Comment:
Could we consider generating something else then just “No” ?

Reply:
No, `Pos` is like metadata that we don't ever compute over except to format into an error message. So just like for `PartialEq` above it's convenient to define identity traits so that the trait derives of types which contain `Pos` still work